packages/ui/src/routes/setup/PasswordSetupPage.tsx

```diff
@@ -45,6 +47,16 @@ const PasswordSetupPage = () => {
     const [passwordScore, setPasswordScore] = useState<number>(0)
     const [isCreating, setIsCreating] = useState<boolean>(false)
 
+    // if the onboarding is ready the user shoulnd't set the password again.
+    const state = useBlankState()
+    useEffect(() => {
+        if (!state || state.isOnboarded) {
```

Comment:
This code is replicated in 3 different files, can we just generalize it? Maybe creating a new hook for instance `useCheckUserIsOnboarded()` or something like that.

Reply:
What do you think?